### ISSUE TITLE
FilterManager causes extra borders

### ISSUE THREAD

**Issue by Maxwell175** [panda3d/panda3d#1227]
## Description
When using the FilterManager, by default the image is in the bottom left with borders on the top and right. This is fixed by setting textures-power-2 to none, in order to prevent the inner texture from being resized incorrectly.

## Steps to Reproduce
```
from panda3d.core import *
from direct.showbase.ShowBase import ShowBase
from direct.filter.FilterManager import FilterManager


# Uncomment the following line to "fix" the problem.
#loadPrcFileData('', "textures-power-2 none")

base = ShowBase()

environ = loader.loadModel("models/world")
environ.reparentTo(render)

base.cam.setPos(0, 0, 10)

filterMgr = FilterManager(base.win, base.cam)
filterQuad = filterMgr.renderSceneInto()
filterQuad.setColor(1, 1, 1, 1)

base.run()
```
If the above sample code is run in the samples/roaming-ralph (just to have some assets to illustrate the issue), you get the following result:
![image](https://user-images.githubusercontent.com/1660330/147797946-dc5b9203-fa75-4e7d-813e-13124c157883.png)

If you un-comment the noted line, the issue is "fixed".

## Environment
 * Operating system: Windows or Linux
 * System architecture: x64
 * Panda3D version:  1.10.10
 * Installation method: (eg. pip, wheel, SDK, built from source) SDK
 * Python version (if using Python): 3.7
 * Compiler (if using C++):

**Comment by rdb** [comment]
This is a feature, not a bug; FilterManager simply expects that you correct this in the shader.

That said, we could at least be writing out correct texture coordinates for the quad.

**Comment by Maxwell175** [comment]
Yeah, I suspected that's the case. I just feel like having it "work"  by default might save someone a trip to the forum or chat.

**Comment by Clockwork-Muse** [comment]
... For that matter, one of the things UnrealEngine does for a similar feature is provide just a single triangle that extends outside the window, mapping the square internal section to the `[0, 1)` range (by mapping the entire triangle into the `[0, 2)` range), and then provide a single shader via a couple of macros (for subwindows, pixel vs uv coordinates, a couple of other cases).  

For almost all use-cases, people don't want the quad, they just want to supply a pixel shader.  This might be more part of the upcoming shader pipeline feature, but a function like: 
```python
class FilterManager():
    ...
    def apply_shader_stage(self, pixel_shader_file: Union[str, Filename], inputs: Optional[Mapping[str, Any]] = None):
        # Apply a shader stage to the current stack
        # Takes a reference to a file containing a pixel shader to apply over the range [0, 1)
```
... would greatly simplify things.

**Comment by Moguri** [comment]
This would be nice to address, but it's probably not happening for 1.11 (PRs are always welcome); moving to 1.12.

**Comment by rdb** [comment]
FWIW I realised doing this via quad coordinates won't work because there can be more than one texture input with different texture resolutions.

Better to document the behaviour, maybe also provide a GLSL equivalent for the texpad input.
